src/actions/AddPeriodicControllers.C

```diff
@@ -0,0 +1,395 @@
+        {
+          _enable_end_start_time_index[j] =
+              (_start_time + _period) + _cycles_per_controls * _period * i + (_period * 0.0001);
+          _enable_end_end_time_index[j] =
+              (_start_time + _period) + _cycles_per_controls * _period * i + (_period * 0.0002);
+        }
+      }
+      else
+      {
+        for (MooseIndex(_enable_end) j = 0; j < _enable_end.size(); ++j)
+        {
+          _enable_end_start_time_index[j] =
+              (_start_time + _period) + _cycles_per_controls * _period * i;
+          _enable_end_end_time_index[j] =
+              (_start_time + _period) + _cycles_per_controls * _period * i + (_period * 0.0001);
+        }
+      }
+
+      for (MooseIndex(_disable_start) j = 0; j < _disable_start.size(); ++j)
+      {
+        _disable_start_start_time_index[j] = _start_time + _cycles_per_controls * _period * i;
+        _disable_start_end_time_index[j] =
+            _start_time + _cycles_per_controls * _period * i + (_period * 0.0001);
+      }
+
+      for (MooseIndex(_disable_during) j = 0; j < _disable_during.size(); ++j)
+      {
+        _disable_during_start_time_index[j] = _start_time + _cycles_per_controls * _period * i;
+        _disable_during_end_time_index[j] =
+            (_start_time + _period) + _cycles_per_controls * _period * i + (_period * 0.0001);
+      }
+
+      // If control cycles are back to back, then disable_start and disable_end times would be the
+      // same, To avoid this, if cycles are back to back, then the disable_end starts at +
+      // 0.0001*period
+      if (_cycles_per_controls == 1.0)
+      {
+        for (MooseIndex(_disable_end) j = 0; j < _disable_end.size(); ++j)
+        {
+          _disable_end_start_time_index[j] =
+              (_start_time + _period) + _cycles_per_controls * _period * i + (_period * 0.0001);
+          _disable_end_end_time_index[j] =
+              (_start_time + _period) + _cycles_per_controls * _period * i + (_period * 0.0002);
+        }
+      }
+      else
+      {
+        for (MooseIndex(_disable_end) j = 0; j < _disable_end.size(); ++j)
+        {
+          _disable_end_start_time_index[j] =
+              (_start_time + _period) + _cycles_per_controls * _period * i;
+          _disable_end_end_time_index[j] =
+              (_start_time + _period) + _cycles_per_controls * _period * i + (_period * 0.0001);
+        }
+      }
+
+      // Adding the multiple TimePeriod controllers
+      if (_enable_start.size() > 0)
+      {
+        if (i == 0)
+        {
+          InputParameters params = _factory.getValidParams("TimePeriod");
+          params.set<std::vector<std::string>>("enable_objects") = _enable_start;
+          params.set<std::vector<Real>>("start_time") = _enable_start_start_time_index;
+          params.set<std::vector<Real>>("end_time") = _enable_start_end_time_index;
+          params.set<ExecFlagEnum>("execute_on", true) = {EXEC_INITIAL, EXEC_TIMESTEP_BEGIN};
```

Comment:
Is there any way to optimize these if statements a bit through this section of the code? It seems as though you are setting the same parameters for the same values for `if true` and then `else` (like `execute_on` and `set_sync_times`). I would pull common set points out one level so that you're doing less repetitive work, and it makes readability better since you immediately see the difference in the cases.